sdks/python/apache_beam/runners/portability/expansion_service_test.py

```diff
@@ -0,0 +1,155 @@
+  def expand(self, pcoll):
+    return pcoll | beam.Map(lambda x: 'Simple(%s)' % x)
+
+  def to_runner_api_parameter(self, unused_context):
+    return 'simple', None
+
+  @staticmethod
+  def from_runner_api_parameter(unused_parameter, unused_context):
+    return SimpleTransform()
+
+
+@ptransform.PTransform.register_urn('multi', None)
+class MutltiTransform(ptransform.PTransform):
+  def expand(self, pcolls):
+    return {
+        'main':
+            (pcolls['main1'], pcolls['main2'])
+            | beam.Flatten()
+            | beam.Map(lambda x, s: x + s,
+                       beam.pvalue.AsSingleton(pcolls['side'])),
+        'side': pcolls['side'] | beam.Map(lambda x: x + x),
+    }
+
+  def to_runner_api_parameter(self, unused_context):
+    return 'multi', None
+
+  @staticmethod
+  def from_runner_api_parameter(unused_parameter, unused_context):
+    return MutltiTransform()
+
+
+@ptransform.PTransform.register_urn('payload', bytes)
+class PayloadTransform(ptransform.PTransform):
+  def __init__(self, payload):
+    self._payload = payload
+
+  def expand(self, pcoll):
+    return pcoll | beam.Map(lambda x, s: x + s, self._payload)
+
+  def to_runner_api_parameter(self, unused_context):
+    return b'payload', self._payload.encode('ascii')
+
+  @staticmethod
+  def from_runner_api_parameter(payload, unused_context):
+    return PayloadTransform(payload.decode('ascii'))
+
+
+@ptransform.PTransform.register_urn('fib', bytes)
+class FibTransform(ptransform.PTransform):
+  def __init__(self, level):
+    self._level = level
+
+  def expand(self, p):
+    if self._level <= 2:
+      return p | beam.Create([1])
+    else:
+      a = p | 'A' >> beam.ExternalTransform(
+          'fib', str(self._level - 1).encode('ascii'),
+          expansion_service.ExpansionServiceServicer())
+      b = p | 'B' >> beam.ExternalTransform(
+          'fib', str(self._level - 2).encode('ascii'),
+          expansion_service.ExpansionServiceServicer())
+      return (
+          (a, b)
+          | beam.Flatten()
+          | beam.CombineGlobally(sum).without_defaults())
+
+  def to_runner_api_parameter(self, unused_context):
+    return 'fib', str(self._level).encode('ascii')
+
+  @staticmethod
+  def from_runner_api_parameter(level, unused_context):
+    return FibTransform(int(level.decode('ascii')))
+
+
+def main(unused_argv):
+  parser = argparse.ArgumentParser()
+  parser.add_argument('-p', '--port',
+                      type=int,
+                      help='port on which to serve the job api')
+  options = parser.parse_args()
+  server = grpc.server(futures.ThreadPoolExecutor(max_workers=2))
+  beam_expansion_api_pb2_grpc.add_ExpansionServiceServicer_to_server(
+      expansion_service.ExpansionServiceServicer(PipelineOptions()), server
+  )
+  server.add_insecure_port('localhost:{}'.format(options.port))
+  server.start()
+  try:
+    while True:
+      logging.info('Listening for expansion requests at %d', options.port)
+      time.sleep(300)
```

Comment:
I don't understand why we need this loop. Either we should timeout after 300 seconds or block here forever.